src/chatgpt-api.ts

```diff
@@ -46,47 +47,19 @@ export class ChatGPTAPI {
    * @param upsertMessage - Optional function to insert or update a message. If not provided, the default implementation will be used (using an in-memory `messageStore`).
    * @param fetch - Optional override for the `fetch` implementation to use. Defaults to the global `fetch` function.
    */
-  constructor(opts: {
-    apiKey: string
-
-    /** @defaultValue `'https://api.openai.com'` **/
-    apiBaseUrl?: string
-
-    /** @defaultValue `false` **/
-    debug?: boolean
-
-    completionParams?: Partial<
-      Omit<types.openai.CreateChatCompletionRequest, 'messages' | 'n'>
-    >
-
-    systemMessage?: string
-
-    /** @defaultValue `4096` **/
-    maxModelTokens?: number
-
-    /** @defaultValue `1000` **/
-    maxResponseTokens?: number
-
-    messageStore?: Keyv
-    getMessageById?: types.GetMessageByIdFunction
-    upsertMessage?: types.UpsertMessageFunction
-
-    fetch?: types.FetchFn
-  }) {
-    const {
-      apiKey,
-      apiBaseUrl = 'https://api.openai.com',
-      debug = false,
-      messageStore,
-      completionParams,
-      systemMessage,
-      maxModelTokens = 4096,
-      maxResponseTokens = 1000,
-      getMessageById = this._defaultGetMessageById,
-      upsertMessage = this._defaultUpsertMessage,
-      fetch = globalFetch
-    } = opts
-
+  constructor({
```

Comment:
the reason I was using `opts` instead of destructuring directly in the constructor is because for some reason in the past `typedoc` wasn't picking up the correct types to document when I did things this way.

will play w/ `typedoc` locally after merging this to see if that's still the case :)